core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1392,17 +1378,7 @@ class DAGScheduler(
 
       if (filesLost || !env.blockManager.externalShuffleServiceEnabled) {
         logInfo("Shuffle files lost for executor: %s (epoch %d)".format(execId, currentEpoch))
-        // TODO: This will be really slow if we keep accumulating shuffle map stages
-        for ((shuffleId, stage) <- shuffleIdToMapStage) {
-          stage.removeOutputsOnExecutor(execId)
-          mapOutputTracker.registerMapOutputs(
-            shuffleId,
-            stage.outputLocInMapOutputTrackerFormat(),
-            changeEpoch = true)
-        }
-        if (shuffleIdToMapStage.isEmpty) {
-          mapOutputTracker.incrementEpoch()
```

Comment:
I think the idea behind this branch was to ensure that we always increment the `mapOutputTracker` epoch upon executor lost. Thus I put an `incrementEpoch()` into the `removeOutputsOnExecutor()` call itself inside `MapOutputTrackerMaster`.